specification/cognitiveservices/OpenAI.Inference/models/images.tsp

```diff
@@ -0,0 +1,122 @@
+
+namespace Azure.OpenAI;
+
+@doc("The desired size of the generated images. Must be one of 256x256, 512x512, or 1024x1024.")
+enum ImageSize {
+    `256x256`,
+    `512x512`,
+    `1024x1024`,
+}
+
+@doc("Defines the request to create an operation to generate images.")
+model ImageRequest {
+    @doc("A text description of the desired image(s).")
+    prompt: string;
+
+    @doc("The number of images to generate.")
+    n: int32;
+
+    @doc("The desired size of the generated images. Must be one of 256x256, 512x512, or 1024x1024.")
+    size: ImageSize;
+
+    @doc("A unique identifier representing your end-user, which can help to monitor and detect abuse.")
+    user: string;
+}
+
+@doc("The image url if successful, and an error otherwise.")
+model ImageResult {
+    @doc("The URL that provides temporary access to download the generated image.")
+    url: string;
+
+    @doc("The error if the operation failed.")
+    error: Foundations.Error;
+}
+
+@doc("The result of the operation if the operation succeeded.")
+model ImageResponse {
+    @doc("A timestamp when this job or item was created (in unix epochs).")
+    created: uint64;
+
+    @doc("The images generated by the operator.")
+    data: ImageResult[];
+}
+
+@doc("The result of the operation if the operation succeeded.")
+model ImageOperationResponse {
+    @doc("The ID of the operation.")
+    id: string;
+
+    @doc("A timestamp when this job or item was created (in unix epochs).")
+    created: int64;
+
+    @doc("A timestamp when this operation and its associated images expire and will be deleted (in unix epochs).")
+    expires: int64;
+
+    @doc("The result of the operation if the operation succeeded.")
+    result: ImageResponse;
+
+    @doc("The status of the operation")
+    status: State;
+}
+
+@lroStatus
+@doc("The state of a job or item.")
+enum State {
+    @doc("The operation was created and is queued to be processed in the future.")
+    "notRunning",
+
+    @doc("The operation has started to be processed.")
+    "running",
+
+    @lroSucceeded
+    @doc("The operation has successfully be processed and is ready for consumption.")
+    "succeeded",
+
+    @lroCanceled
+    @doc("The operation has been canceled and is incomplete.")
+    "canceled",
+
+    @lroFailed
+    @doc("The operation has completed processing with a failure and cannot be further consumed.")
+    "failed",
+
+    @doc("The entity has been deleted but may still be referenced by other entities predating the deletion.")
+    "deleted",
+}
+
+@doc("Provides status details for long running operations.")
+model ImageOperationStatus {
+  @key("operationId")
+  @visibility("read")
+  @doc("The unique ID of the operation.")
+  id: string;
+
+  @doc("The status of the operation")
+  status: State;
+}
+
+#suppress "@azure-tools/typespec-azure-core/long-running-polling-operation-required" "template"
+@Foundations.Private.needsRoute
+@post
+@doc("Long running operation template to match Azure OpenAI .")
```

Comment:
Does this doc end up in emitted ref docs or are we purely limited to custom code for that? This doesn't seem user-facing as-is but that may be expected.

Reply:
from what I could see, this type was not generated as is, so the docs wouldn't show up to the developer customer